KubeOne: documentation for Operating System Manager (#1133)
kubermatic/docs: content/kubeone/master/guides/operating-system-manager/usage/_index.en.md

```yaml
apiVersion: kubeone.k8c.io/v1beta2
kind: KubeOneCluster
versions:
  kubernetes: 1.23.6
addons:
  enable: true
operatingSystemManager:
  deploy: false
```

## Migrating existing machine deployments

{{% notice warning %}}
Existing Machine Deployments will not be rotated automatically. Existing machine deployment
should be rotated manually, for them to consume configurations generated by OSM.
{{% /notice %}}

To perform this rotation please follow the guide at [Rolling Restart MachineDeploments][rolling-restart].

[rolling-restart]: {{< ref "../../../cheat_sheets/rollout_machinedeployment" >}}

## Using custom OperatingSystemProfile

The annotation `k8c.io/operating-system-profile` is used at the Machine Deployment level to select the Operating System Profile that will be used to generate configurations for provisioning the machines.

### When creating a new cluster

For new clusters, a dedicated terraform variable `initial_machinedeployment_operating_system_profile` was introduced for this purpose. It'll propagate the specified value to the initial machine deployments in the cluster.

It can be used as follows:

> 1. Create a custom OSP and place it in a folder inside your custom addons path, if it exists otherwise create one. Let's take this minimalistic OSP just for the sake of understanding:

Reviewer: ```suggestion
1. Create a custom OSP and place it in a folder inside your custom addons path, if it exists, otherwise create one. Let's take this minimalistic OSP as an example:
```

kind: KubeOneCluster
versions:
  kubernetes: "1.22.5"
cloudProvider:
  aws: {}
addons:
  enable: true
  path: "./custom-addons"
```

**NOTE:** The path for addons can be anything. But make sure that the custom OSP exists in that directory.

### For an existing machine deployments

For existing Machine Deployments; edit the machine deployment and update `k8c.io/operating-system-profile` annotation with the required OSP name.

{{% notice warning %}}
Secrets and OSC against the Machine Deployment will be rotated right away when we update the annotation. But the existing Machine Deployments will not be rotated automatically.
Existing machine deployment should be rotated manually, for them to consume configurations generated by OSM.
{{% /notice %}}

To perform this rotation for existing Machine Deployments please follow the guide at [Rolling Restart MachineDeploments][rolling-restart].

[rolling-restart]: {{< ref "../../../cheat_sheets/rollout_machinedeployment" >}}

### For new machine deployments

Create a new Machine Deployment and make sure that you add the `k8c.io/operating-system-profile` annotation with the required OSP name.

## Additional Information

> - If an `OperatingSystemProfile` is updated, that wouldn't result in an automatic rotation of the machines. This is an intentional design decision since an `OperatingSystemProfile` can be associated with multiple Machine Deployments and it is the user's responsibility to rotate the machines when the `OperatingSystemProfile` is updated.

Reviewer: ```suggestion
- An update to `OperatingSystemProfile` will not result in an automatic rotation of the machines. This is an intentional design decision since an `OperatingSystemProfile` can be associated with multiple Machine Deployments. It is the user's responsibility to rotate the machines when the `OperatingSystemProfile` is updated.
```
